src/components/modules/readonly.ts

```diff
@@ -0,0 +1,104 @@
+import Module from '../__module';
+import * as _ from '../utils';
+
+/**
+ * @module ReadOnly
+ *
+ * Has one important method:
+ *    - {Function} toggleReadonly - Set read-only mode or toggle current state
+ *
+ * @version 1.0.0
+ *
+ * @typedef {ReadOnly} ReadOnly
+ * @property {boolean} readOnlyEnabled - read-only state
+ */
+export default class ReadOnly extends Module {
+  /**
+   * Value to track read-only state
+   *
+   * @type {boolean}
+   */
+  private readOnlyEnabled = false;
+
+  /**
+   * Returns state of read only mode
+   */
+  public get isEnabled(): boolean {
+    return this.readOnlyEnabled;
+  }
+
+  /**
+   * Set initial state
+   */
+  public async prepare(): Promise<void> {
+    if (!this.isToolsReadOnlyCompatible()) {
+      return;
```

Comment:
also, the editor should not be initialized if readOnly enabled, but some tool does not support it. I mean it should throw an error and do not render anything.